sys/freebsd/kqueue.txt

```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <sys/types.h>
+include <sys/event.h>
+
+kqueue() fd
+kevent(fd fd, changelist ptr[in, array[kevent]], nchanges len[changelist], eventlist ptr[in, array[kevent]], nevents len[eventlist], timeout ptr[in, timespec])
+
+kevent {
+	ident	intptr
+	filter	flags[filters, int16]
+	flags	flags[evflags, int16]
+	fflags	flags[fflags, int32]
+	data	int64
+	udata	intptr
+	ext	array[int64, 4]
+}
```

Comment:
So in event.h you'll notice that we also have a compatibility structure, kevent_freebsd11. Correspondingly, in syscalls.master there is a COMPAT11 definition for kevent() using struct kevent_freebsd11.

This basically means that during the FreeBSD 11 development cycle we changed the layout of struct kevent. These compat structures exist to allow software compiled using the old struct kevent layout to continue working on FreeBSD without a recompile.

It's important to fuzz these compat system calls as well since the kernel code which handles them by converting from the freebsd11 kevent layout to the current layout can be and frequently is a source of bugs. If you search for "freebsd11" and "freebsd12" in the FreeBSD syscall descriptions you'll see some examples of how to cover compat syscalls.